### PR TITLE
docs: rephrase and clarify plugin rules

### PR DESCRIPTION
**Pull request by bastimeyer**
See #5904
Follow-up of #5947 

This doesn't update any of the current plugin rules and just rephrases and clarifies them.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -45,31 +45,64 @@ Feature requests are welcome, but take a moment to find out whether your idea fi
...
+Plugin submissions and requests are a great way to improve Streamlink. Requests should be as detailed as possible and dedicated to only a single streaming website/service.
 
-Plugins which fall under the following categories will not be implemented or considered and the request will be closed:
+Information about the website, as well as explicit URLs from your web browser's URL address bar are required, as are details regarding the website, what it is, who it is run by, and more. This allows for easier plugin review and implementation. We reserve the right to refuse to implement, develop, or update any plugin. In addition, we may remove existing plugins at our own discretion at any time without reasoning.
```
Comment (gravyboat) [inline CONTRIBUTING.md:50]: > we may remove existing plugins at our own discretion at any time without reasoning.

maybe instead say:

> we may remove existing plugins at our own discretion at any time without explanation.

Some people may try to argue with us on this point if they believe we can be reasoned with.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -45,31 +45,64 @@ Feature requests are welcome, but take a moment to find out whether your idea fi
...
+1. **Any kind of streaming service that uses DRM protection**
 
-8. Sites where the maintainer has requested we not add their site to Streamlink
+    Streamlink won't implement or break any DRM schemes.
 
-9. Sites which are unmaintained, are in beta or are undergoing heavy amounts of development and may change rapidly
+2. **Sites which are hosting stolen content as their main source of content**
 
-10. Sites which have no way to determine viewership numbers
+    Plugins must only support websites with authentic content.
```
Comment (gravyboat) [inline CONTRIBUTING.md:69]: How are we defining "authentic" here? This was an annoying spot we got in to in the past when people asked us to implement some weird TV station that was running news broadcasts or other local content. Does this need to be more explicit? I don't think a nightly news broadcast for a town of 20 people really counts as authentic or non-niche.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -45,31 +45,64 @@ Feature requests are welcome, but take a moment to find out whether your idea fi
...
+
+    Plugins must only implement the original/primary website where the content is hosted or shown.
+
+4. **Sites which require paid logins or subscriptions**
+
+    Plugin implementations and maintenance must be possible by everyone. Paid access prevents this.
+
+5. **NSFW sites of a pornographic nature**
+
+    This also includes borderline NSFW content, as it lowers the bar for follow-up plugin requests.
```
Comment (gravyboat) [inline CONTRIBUTING.md:81]: 👍 Good addition to note here regarding borderline content.

**Review by gravyboat — APPROVED** [review]
Really good changes. I've added a minor suggestion and brought up one point around content which could put us in a sticky spot regarding which might allow people to start asking for their local TV station websites again. Take or leave whatever makes sense to you.

File: CONTRIBUTING.md (hunk 2 above)
Comment (bastimeyer) [reply CONTRIBUTING.md:69]: I believe you're confusing two rules here. Niche content is already covered by rule 10.

Comment (gravyboat) [reply CONTRIBUTING.md:69]: No I'm aware of rule 10 but my assumption is most people aren't going to read all the rules so I'm not sure if we should be more explicit in this one. It's not a major deal or anything, I just don't want the headache of arguing with people about what the word "authentic" means so I figured I'd mention it. We can revise later if it becomes a problem.

**Pull request merged**